pkg/adapter/metrics.go

```diff
@@ -69,7 +69,19 @@ type (
 		Builder
 
 		// NewMetrics returns a new instance of the Metrics aspect.
-		NewMetrics(env Env, config AspectConfig) (MetricsAspect, error)
+		NewMetrics(env Env, config AspectConfig, metrics []MetricDefinition) (MetricsAspect, error)
+	}
+
+	// MetricDefinition provides the basic description of a metric schema
+	// for which metrics adapters will be sent Values at runtime.
+	MetricDefinition struct {
+		// Name is the canonical name of the metric.
+		Name string
+		// Kind provides type information about the metric.
+		Kind Kind
+		// Labels are the names of keys for dimensional data that will
+		// generated at runtime and passed along with metric values.
+		Labels []string
```

Comment:
will BE generated...

Wouldn't it be desirable to know the type of the label values? When talking to GCM for example, I think it likes to know when things are integers/floats rather than just strings. This could be a map[string]config.AttributeDescriptor.ValueType (or whatever that enum is called in Go...). Maybe that map deserves to be its own type (LabelMap?] and then we can use it elsewhere like in QuotaDefinition.